src/test_utils.rs

```diff
@@ -0,0 +1,40 @@
+//! Internal utilities to support testing.
+
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::layer::{Layer, SubscriberExt};
+use tracing_subscriber::util::SubscriberInitExt;
+
+/// Configures logging for the current test.
+///
+/// This configures logging that will interact well with tests: logs will be captured by the
+/// test framework and only printed on failure.
+///
+/// The logging is unfiltered, it logs all crates and modules on TRACE level.  If that's too
+/// much consider if your test is too large (or write a version that allows filtering...).
+#[must_use = "The tracing guard must only be dropped at the end of the test"]
+pub(crate) fn setup_logging() -> tracing::subscriber::DefaultGuard {
+    let log_layer = tracing_subscriber::fmt::layer()
+        .with_writer(|| TestWriter)
+        .with_filter(LevelFilter::TRACE);
```

Comment:
So it is no longer possible to configure log level with RUST_LOG? I find trace to be a bit too much usually...